docs/book/cookbook/aot-guide.md

```diff
@@ -0,0 +1,112 @@
+# Using AoT In A ZendFramework App
+
+This guide will show you how you can use zend-di's AoT compiler
+to make your ZendFramework application production ready.
+
+You will learn how to:
+
+* Add a script to run the compilation
+* Use the generated injector with ServiceManager
+* Use the generated factories with ServiceManager
+
+## Create project and add zend-di
+
+For this guide will use an [expressive application](https://docs.zendframework.com/zend-expressive/)
+with zend-servicemanager as example.
+
+If you already set up a project with zend-di, you can skip this step.
+
+```bash
+composer create-project zendframework/zend-expressive-skeleton zend-di-aot-example
+```
+
+Pick the components you want to use. As statet above, we will be using Zend ServiceManager
+and a Modular layout for this example.
+
+Now add zend-di with composer:
+
+```bash
+composer require zendframework/zend-di=^3.0
```

Comment:
Why the version constraint?

Reply:
I thought it states clearly that this requires 3.0 at minimum and prevents composer to accidentally install zend-di 2.x.
Should there be more explanation about this?

Reply:
If version 3.0 is released, then version 3 will be installed automatically. The version constraint is not needed. Remove it and all is fine.
We should not over complicated the documentation.

Reply:
I agree with both of you!

I'd suggest demonstrating _without_ the version constant, but adding a note that if v2 gets installed, you should try the above (with constraint) to determine what is blocking it. You might also want to consider adding some verbiage in that note that if you are upgrading from v2, you will need to remove zend-servicemanager-di first.

Reply:
Good points.
I totally missed the zend-servicemanager-di conflict. I'll add a Note and hint to the version constraint option as sugessted by @weierophinney

Reply:
For version hint: 👍